core/templates/services/html-length.service.ts

```diff
@@ -16,85 +16,210 @@
+        textContent, calculationType);
     }
-    return totalWeight;
+    return totalLength;
   }
 
-  private getWeightForTextNodes(textNode: HTMLElement): number {
-    const textContent = textNode.textContent || '';
-    const words = textContent.trim().split(' ');
-    const wordCount = words.length;
+  /**
+    * This function calculates the length of a given text content
+    * based on the specified calculation type.
+    *
+    * @param {string} textContent - The text content for which the
+    *    length is to be calculated. This is typically a string of
+    *    text that may include words, spaces, punctuation, and other
+    *    characters.
+    * @param {CalculationType} calculationType - The method used to
+    *    calculate the length. It can be either 'word' or 'character'.
+    *    If 'word', the function counts the number of words in the
+    *    text content. If 'character', it counts the number of
+    *    characters.
+    * @returns {number} The length of the text content. This is
+    *    calculated as the number of words or characters in the text
+    *    content, depending on the calculation type.
+    */
 
-    return wordCount;
+  private calculateTextLength(
+      textContent: string, calculationType: CalculationType): number {
+    let trimmedTextContent = textContent.trim();
+    let totalLength = 0;
+    if (calculationType === CALCULATION_TYPE_WORD && trimmedTextContent) {
+      const words = trimmedTextContent.split(' ');
+      totalLength = words.length;
+    } else {
+      totalLength = trimmedTextContent.length;
+    }
+    return totalLength;
   }
 
-  private getWeightForNonTextNodes(nonTextNode: string): number {
-    if (nonTextNode.includes('oppia-noninteractive-math')) {
-      return 1;
-    }
-    const altTextMatch = nonTextNode.match(/alt-with-value=["']([^"']*)["']/);
-    let words = [];
-    if (altTextMatch && altTextMatch[1]) {
-      const altText = altTextMatch[1];
-      words = altText.trim().split(' ');
+  /**
+    * This function calculates the length of a given non-text
+    * node based on the specified calculation type.
+    *
+    * @param {string} nonTextNode - The non-text node for which
+    *    the length is to be calculated. This is typically an HTML
+    *    string that represents a non-text element in the DOM,
+    *    such as an image or a video element.
+    * @param {CalculationType} calculationType - The method used
+    *    to calculate the length. It can be either 'word' or
+    *    'character'. If 'word', the function counts the number
+    *    of words in the non-text node. If 'character', it counts
+    *    the number of characters.
+    * @returns {number} The length of the non-text node. This is
+    *    calculated as the number of words or characters in the
+    *    non-text node, depending on the calculation type.
+    */
+
+  // TODO(#19729): Create RTE-component-specific logic
+  // for calculating the lengths of RTE-components.
```

Comment:
I'm fine with adding this TODO for now given that this PR unblocks users but can you please assign the task to yourself so that it can be worked on after this PR goes in? Thanks.